nautobot/extras/tests/test_dynamicgroups.py

```diff
@@ -410,17 +411,29 @@ def test_remove_child(self):
     def test_generate_query_for_filter(self):
         """Test `DynamicGroup.generate_query_for_filter()`."""
         group = self.parent  # Any group will do, so why not this one?
-        value = ["site-3"]
-        query = group.generate_query_for_filter(
-            filter_field=group.filterset_class.declared_filters["site"],
-            value=value,
+        multi_value = ["site-3"]
+        fs = group.filterset_class()
+        multi_field = fs.filters["site"]
+        multi_query = group.generate_query_for_filter(
+            filter_field=multi_field,
+            value=multi_value,
         )
 
+        queryset = group.get_queryset()
+
         # Assert that both querysets resturn the same results
-        group_qs = group.get_queryset().filter(query)
-        device_qs = Device.objects.filter(site__slug__in=value)
+        group_qs = queryset.filter(multi_query)
+        device_qs = Device.objects.filter(site__slug__in=multi_value)
         self.assertQuerySetEqual(group_qs, device_qs)
 
+        # Now do a non-multi-value filter
+        solo_field = fs.filters["serial"]
```

Comment:
Add a note here: If we ever make `serial` a multi-value filter, this will need to be revised.